cluster/juju/layers/kubernetes-master/config.yaml

```diff
@@ -41,6 +41,12 @@ options:
     default: ""
     description: |
       Password to be used for admin user (leave empty for random password).
+  grafana-www-root:
+    type: string
+    default: "/api/v1/proxy/namespaces/kube-system/services/monitoring-grafana/"
+    description: |
+      Set this to "/" if you plan to expose the grafana used for cluster monitoring.
+      The default value is for accessing the service through kubectl proxy.
```

Comment:
Do we need this? I'm worried about accumulating a lot of configuration options that we can't keep up with.

We want to empower users to set things up however they like, but I'm not sure this particular config is worth the maintenance burden, considering they have another option: set `enable-dashboard-addons=false` and deploy their own Grafana.

If we can, I would rather say that the Grafana shipped with CDK is only available via the apiserver at `/api/v1/proxy/namespaces/kube-system/services/monitoring-grafana/`.

Reply:
This is a fair point @Cynerva. I mentioned in our stand-up this PR as a proposal on how we could address the issue. Being opinionated on how self hosted monitoring works is fine by me. I would also take it a step further and question why would we have this monitoring as part of the charms given that a) we have a spell for monitoring CDK externally (Prometheus+graylog) and b) the user can deploy his own self hosted add-on. A spell for self-hosted monitoring might simplify things. The complexity the cdk-addons snap is adding to our code and release processes makes me wonder if we should try to get rid of it. But that is a story for another time.

For this PR lets get feedback from @tvansteenburgh , @wwwtyro , @kwmonroe , @hyperbolic2346 so we are all on the same page. Also if we decide on supporting only one of the access paths what would that be?
1. For kubectl proxy needs the path to be /api/v1/proxy/namespaces/kube-system/services/monitoring-grafana/ and the service is available at http://localhost:8001/api/v1/proxy/namespaces/kube-system/services/monitoring-grafana/?orgId=1&editview=settings
2. To access grafana using the URL you get from kubectl cluster-info the path should be  /api/v1/namespaces/kube-system/services/monitoring-grafana/proxy. This might be a bug with the grafana yaml
https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/influxdb/influxdb-grafana-controller.yaml#L75
not being aligned with how cluster-info builds the URLs
https://github.com/liggitt/kubernetes/commit/9262c57ba8409fa3b2b91a3af56df6243446abe3
3. Set the path to / so users can expose the deployment.

Reply:
My vote is for excluding this change for the reasons @Cynerva outlined, and for moving grafana out of the charms for the reasons @ktsakalozos outlined. If we have anything of this nature, it should be an action or a spell.

Reply:
@ktsakalozos I agree with @wwwtyro (and @Cynerva by proxy). Rather than complicating `kubernetes-master` with config specific to grafana, I would prefer we handle `<dashboard>` in a self-contained action or spell addon.

Reply:
I'm +1 on removing grafana from cdk-addons.